lib/membership/membership-router-spec.js

```diff
@@ -0,0 +1,218 @@
+  var mockTopics = function(mediator) {
+    this.mediator = mediator;
+  };
+  mockTopics.prototype.request = function() {};
+  var mockTopicRequest = sinon.stub(mockTopics.prototype, "request", mockMembershipCloudDataTopics.request);
+
+  //Mock Data
+  var req = {};
+  var res = {};
+  var mockErr;
+  var mockMembershipList = [
+    {id: "membership-test-id-1", group: 'test-group-1', user: 'test-user-1'},
+    {id: "membership-test-id-2", group: 'test-group-2', user: 'test-user-2'},
+    {id: "membership-test-id-3", group: 'test-group-3', user: 'test-user-3'}
+  ];
+  var membershipIdSuc = 'membership-test-id-1';
+  var membershipIdNA = 'membership-test-id-NA';
+  var membershipIdErr = 'membership-test-id-ERR';
+
+  beforeEach(function(done) {
+    mockTopicRequest.reset();
+    app = express();
+    app.use(bodyParser.json());
+    request = supertest(app);
+    membershipRouter = proxyquire('./membership-router.js', {
+      'fh-wfm-mediator/lib/topics': mockTopics
+    });
+    membershipRouter(mediator, app);
+    done();
+  });
+
+  //List
+  it('should return a list of existing memberships', function(done) {
+    request
+      .get('/api/wfm/membership/')
+      .send(req, res)
+      .expect(200, function(err, res) {
+        assert.ok(!res.error, "Expected no response error to be received");
+        assert.ok(res, "Expected a result to be received from the list request");
+        assert.deepEqual(res.body, mockMembershipList, "Expected membership list to be returned from the list request");
+        sinon.assert.calledWith(mockTopicRequest, 'list');
+        sinon.assert.calledOnce(mockTopicRequest);
+        done();
+      });
+  });
+
+  //Read
+  it('should read an existing membership', function(done) {
+    request
+      .get('/api/wfm/membership/' + membershipIdSuc)
+      .send(req, res)
+      .expect(200, function(err, res) {
+        assert.ok(!res.error, "Expected no response error to be received");
+        assert.ok(res, "Expected a result to be received from the read request");
+        assert.deepEqual(res.body, mockMembershipList[0], "Expected membership with the given membership id to be returned");
+        sinon.assert.calledWith(mockTopicRequest, 'read', membershipIdSuc);
+        sinon.assert.calledOnce(mockTopicRequest);
+        done();
+      });
+  });
+
+  it('should return an empty object upon reading a non-existing membership', function(done) {
+    request
+      .get('/api/wfm/membership/' + membershipIdNA)
+      .send(req, res)
+      .expect(200, function(err, res) {
+        assert.ok(!res.error, "Expected no response error to be received");
+        assert.ok(res, "Expected a result to be received from the read request");
+        assert.deepEqual(res.body, {}, "Expected an empty object upon reading a membership that does not exist");
+        sinon.assert.calledWith(mockTopicRequest, 'read', membershipIdNA);
+        sinon.assert.calledOnce(mockTopicRequest);
+        done();
+      });
```

Comment:
`supertest` has a lot of shorthands, so this could be a little more readable as
```javascript
.expect(200, {})
.expect(function() {
//... sinon asserts
})
.end(done)
```